server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java

```diff
@@ -44,7 +50,15 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        final boolean includeTypeName = request.paramAsBoolean(INCLUDE_TYPE_NAME_PARAMETER,
+            DEFAULT_INCLUDE_TYPE_NAME_POLICY);
+        if (request.hasParam(INCLUDE_TYPE_NAME_PARAMETER)) {
+            deprecationLogger.deprecatedAndMaybeLog("index_rollover_with_types", TYPES_DEPRECATION_MESSAGE);
+        }
         RolloverRequest rolloverIndexRequest = new RolloverRequest(request.param("index"), request.param("new_index"));
+        if (includeTypeName == false) {
+            rolloverIndexRequest.setNoTypeNameForMappingParsing();
```

Comment:
This overall approach looks good to me, but I wonder if instead of setting a special flag on the request here, we can just pass a boolean `includeTypeName` to the `fromXContent` method? In particular, `parse` methods take an arbitary `Context` parameter that I think you could use to pass in the boolean. Here is an example from `FieldCapabilities` where we are passing in a `String` name to the `ObjectParser`: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilities.java#L136-L138